### DISCUSSION TITLE
Error : The nonce in ID Token  does not match the SHA256 hash of the raw nonce n the request.

### DISCUSSION THREAD

**Discussion by IanFangQuad** [firebase/flutterfire#10296] [Q&A]
Hello, I am new to flutter.  I want to integrate line login with firebase.
First, I have set line as a auth provider in firebase console with openID connect. 
Second, I use flutter_line_sdk to login user and receive access token , id token, etc. 
Then I want to use these data to get credential by OAuthProvider("oidc.line").credential(....) and signInWithCredential( credential ), but always got this error while the two nonce looks same when get credential, no matter use line generate nonce or use nonce generate by myself. 
Where should I notice? Thanks.
＊test on IOS
<img width="580" alt="image" src="https://user-images.githubusercontent.com/117148221/213412180-407eb160-963b-466a-9793-24bd191e7972.png">
<img width="769" alt="image" src="https://user-images.githubusercontent.com/117148221/213412841-a07aedae-efb5-45bc-9970-2d89fddec873.png">

**Comment by YoamFarges** [comment]
@IanFangQuad did you manage to make it work or to find a solution?
Currently experiencing the same issue.

**Comment by Viraj-Tank** [comment]
@IanFangQuad @YoamFarges I'm also facing the same issue, also tried generating the random string for nonce but its not working at all

**Comment by unbam** [comment]
I had the same issue. Resolved?

**Reply by unbam** [reply]
I resolved.

```dart
final nonce = Nonce.generate();
final digest = sha256.convert(utf8.encode(nonce));
final sha256Nonce = digest.toString();
final loginOption = LoginOption(false, 'normal')..idTokenNonce = sha256Nonce;
final result = await LineSDK.instance.login(
  scopes: ['profile', 'openid', 'email'],
  option: loginOption,
);

final provider = OAuthProvider('oidc.line');
final credential = provider.credential(
  idToken: result.accessToken.idTokenRaw,
  rawNonce: nonce,
);

await _firebaseAuth.currentUser!.linkWithCredential(credential);
```

**Reply by Viraj-Tank** [reply]
Sure, will try this!

**Reply by adityapol7711** [reply]
Hi, I'm facing the same issue while authorizing in firebase through angular and keycloak.
Do you have any idea on how to solve this in angular?

**Comment by efstathiosntonas** [comment]
Just in case another poor soul ends up here from google search: 

The `rawNonce` is the "pure" string without being converted to SHA256 as the `raw` in the name suggests.

This is the equivalent code in react-native: 

```tsx
import * as Crypto from "expo-crypto";

    const nonceRaw = uuid();
    
       const digest = await Crypto.digestStringAsync(
        Crypto.CryptoDigestAlgorithm.SHA256,
        nonceRaw
      );

 const credential = auth.FacebookAuthProvider.credential(
        // @ts-ignore
        isIOS ? fbToken.authenticationToken : fbToken.accessToken,
        isIOS ? nonceRaw : undefined  <---- notice we pass the uuid() here without sha256 applied to it
      );

      const { user, additionalUserInfo } = await auth().signInWithCredential(credential);
```
